tests/flask_integration/flask.py

```diff
@@ -152,3 +159,29 @@ def _confirm_company(
             follow_redirects=True,
         )
         assert response.status_code < 400
+
+    def assert_response_has_expected_code(
```

Comment:
I think that this function is problematic. It takes way to many arguments for what it does and is basically just a wrapper around `self.client.get` and `self.client.post`. Further the `login` argument changes the state of the client in a unexpected way. Before the method call the test client might not be authenticated but after calling this method the client might be authenticated. I find this behavior surprising. Looking at where you used this method it also did not make the code more concise.

```
self.login_member()
assert self.client.get(url).status_code == expected_status_code
```
is more concise than

```
self.assert_response_has_expected_code(
    url=expected_url,
    method='get',
    login=LoginUser.member,
    expected_code=expected_status_code,
)
```

Reply:
I agree with the arguments you made and think that this change should go forward.